### DISCUSSION TITLE
how do I search for a pattern starting with "-"?

### DISCUSSION THREAD

**Discussion by tgray** [BurntSushi/ripgrep#1560] [General] [answered]
#### What version of ripgrep are you using?

macOS homebrew:
ripgrep 12.0.1
-SIMD -AVX (compiled)
+SIMD +AVX (runtime)

also on Windows precompiled:
ripgrep 12.0.1 (rev 1d5b1011e5)
-SIMD -AVX (compiled)
+SIMD +AVX (runtime)

#### How did you install ripgrep?

macOS: homebrew
Windows: compiled binary

#### What operating system are you using ripgrep on?

macOS 10.14.6
Windows 10 something

#### Describe your bug.

Trying to use rg to search for a literal string that starts with a '-', specifically markdown style check boxes '- [ ]'.

#### What are the steps to reproduce the behavior?

    rg -F '- [ ]' .

or 

    rg -F "- [ ]" .

Running the search by escaping the leading dash works fine, but defeats the purpose of using -F in my opinion.

#### What is the actual behavior?

Normal and debug output are the same.

```
rg -F '- [ ]' --debug .
error: Found argument '- ' which wasn't expected, or isn't valid in this context

USAGE:

    rg [OPTIONS] PATTERN [PATH ...]
    rg [OPTIONS] [-e PATTERN ...] [-f PATTERNFILE ...] [PATH ...]
    rg [OPTIONS] --files [PATH ...]
    rg [OPTIONS] --type-list
    command | rg [OPTIONS] PATTERN

For more information try --help
```

#### What is the expected behavior?

rg should have searched the literal string '- [ ]' in the current directory.

**Comment by BurntSushi** [comment, marked as answer]
This is correct and expected behavior. The `-F` flag is a red herring here. If your pattern starts with a `-`, then you must either precede it with the `-e` flag or use `--` to indicate that all subsequent arguments are positional. This is documented in the man page:

> A regular expression used for searching. To match a pattern beginning with a dash, use the -e/--regexp option.

So you'll want `rg -F -e '- [ ]'` or `rg -F -- '- [ ]'`.

> Running the search by escaping the leading dash works fine

This is a different search. There is no escaping when using `-F`, so if you use `\- [ ]`, then you'll actually be searching for a literal `\\` followed by a `-`:

```
$ echo '- [ ]' | rg -F '\- [ ]'
$ echo '\- [ ]' | rg -F '\- [ ]'
\- [ ]
$ echo '- [ ]' | rg -F -e '- [ ]'
- [ ]
```

Note that grep has the same exact behavior here and the exact same work-arounds.

**Comment by BurntSushi** [comment]
Moved this to a discussion, since this is more of a question of behavior even though it was reported as a bug.
